### PR TITLE
feat(django): add option to append url_name of resolver match to span resource name

### PR DESCRIPTION
**Pull request by ca-simone-chiorazzo**
## Description
This PR aims to introduce a backwards compatible change that allows Django <2.2.0 developers to be able to distinguish different spans related to URLs under the same view.

As of today, ddtrace uses a pattern `{method} {handler}` that in case of class-based viewsets doesn't allow the installer to distinguish different endpoints under the same class in a clear way.
 
In the end, the PR adds a new optional configuration to append the url_name of Django resolver match to differentiate endpoints under the same ViewSet.

## Checklist
- [X] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [X] Add additional sections for `feat` and `fix` pull requests.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Motivation
Without this change, using django 1.X, it's not possible to trace different DRF endpoints that are under the same ViewSet. The actual behaviour is that ddtrace aggregates everything that is served under the same ViewSet. 

## Reviewer Checklist
- [X] Title is accurate.
- [X] Description motivates each change.
- [X] No unnecessary changes were introduced in this PR.
- [X] PR cannot be broken up into smaller PRs.
- [X] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [X] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

**Comment by ca-simone-chiorazzo** [comment]
Hi @brettlangdon ,

it would be great if you could give a look at this PR since it would allow us to build the Datadog dashboard that we would like to have for Django 1 application. 

In addition to whatever needs to be fixed, I also ask you for help on the CI part. There is mypy failing on internal modules in which there is no changes mine. Any clue on it?

Thanks!

**Comment by Yun-Kim** [comment]
> In addition to whatever needs to be fixed, I also ask you for help on the CI part. There is mypy failing on internal modules in which there is no changes mine. Any clue on it?
> 
> Thanks!
It looks like mypy just released a new version 0.981 which we automatically use in our mypy check.

**Comment by majorgreys** [comment]
@ca-simone-chiorazzo This will need a release note, thanks!

**Comment by ca-simone-chiorazzo** [comment]
> @ca-simone-chiorazzo This will need a release note, thanks!

Sorry @majorgreys, is there a guide on how to add a release note for the PR? Thanks!

**Comment by ZStriker19** [comment]
Hi @ca-simone-chiorazzo for the release notes, I couldn't find a guide (we're working on improving this) but basically:
running `reno new <file_name_for_note>` will create the release note and tell you where it is. From there you can just reference other release notes for what the file content should look like: https://github.com/DataDog/dd-trace-py/tree/1.x/releasenotes/notes

**Comment by ca-simone-chiorazzo** [comment]
@ZStriker19 @majorgreys Release note added! Let me know what do you think

**Comment by ca-simone-chiorazzo** [comment]
Looks like that there are flaky/broken tests related to psycopc and httpx. What's the procedure in this case? It doesn't seems related to my PR

**Review by ZStriker19 — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]
This feature contribution certainly fits into the solution we currently have in that the integration supports only a limited number of resource names. As has already come up, the design of the Django integration does not give flexibility to a ddtrace user to override the request span's resource name. I would be in favor of revisiting this design and such a constraint on a web integration.

**Comment by ca-simone-chiorazzo** [comment]
Hi folks, I've just noticed that the merge is blocked due to missing signatures on commits. I've set up a GPG signature on my account, but the only way I found to sign commits already pushed is to execute a forced push.

 Do you perhaps have some guide or best practice in order to resign commit already pushed?

File: releasenotes/notes/add-config-to-allow-different-resource-name-in-django-dd55912a96678653.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Django: Add configuration option to allow a resource format like `{method} {handler}.{url_name}` in projects with Django <2.2.0
```
Comment (mabdinur) [inline releasenotes/notes/add-config-to-allow-different-resource-name-in-django-dd55912a96678653.yaml:4]:
```suggestion
    django: add configuration option to allow a resource format like `{method} {handler}.{url_name}` in projects with Django <2.2.0

```

**Comment by ca-simone-chiorazzo** [comment]
https://github.com/Mergifyio backport 0.61

**Comment by ca-simone-chiorazzo** [comment]
Guys I need to backport the change on the 0.61 since the 1.6.0 doesn't work with python 2, I'm getting an error when using
`django1.11`
`django-import-export==1.2.0`

I get the following error
```python
  File "/myproject/users/admin.py", line 8, in <module>
    from import_export import resources, fields
  File "/versions/myvenv/lib/python2.7/site-packages/ddtrace/internal/module.py", line 180, in _load_module
    module = self.loader.load_module(fullname)
  File "/versions/myvenv/lib/python2.7/pkgutil.py", line 243, in load_module
    mod = imp.load_module(fullname, self.file, self.filename, self.etc)
  File "/versions/myvenv/envs/bakery/lib/python2.7/site-packages/import_export/resources.py", line 191, in <module>
    class Resource(six.with_metaclass(DeclarativeMetaclass)):
  File "/versions/myvenv/lib/python2.7/site-packages/django/utils/six.py", line 808, in __new__
    return meta(name, bases, d)
  File "/versions/myvenv/envs/bakery/lib/python2.7/site-packages/import_export/resources.py", line 159, in __new__
    setattr(meta, option, getattr(options, option))
AttributeError: 'NoneType' object has no attribute 'attrs'
```

how can I proceed?

**Comment by Kyle-Verhoog** [comment]
hey @ca-simone-chiorazzo, thanks for reporting! Are you seeing this in ddtrace 1.5 as well? We have a similar issue we're looking into (https://github.com/DataDog/dd-trace-py/issues/4484) (released in 1.6) which might be related.

**Comment by ca-simone-chiorazzo** [comment]
Hi @Kyle-Verhoog , thanks for getting back to me. The issue is not present in ddtrace <=1.5.0, so it's something broken on 1.6.0

**Comment by ca-simone-chiorazzo** [comment]
@Kyle-Verhoog Let me know if you need you more information about the issue

**Comment by Kyle-Verhoog** [comment]
hi @ca-simone-chiorazzo I'm pretty confident that the issue is #4484. We have a fix coming for it in 1.6.1 to be released today hopefully!

**Pull request merged**